rpcd/playbooks/rpc-pre-upgrades.yml

```diff
@@ -19,18 +19,60 @@
   roles:
     - { role: "rpc_pre_upgrade", tags: [ "rpc-pre-upgrades" ] }
 
-- name: restart beaver
+- name: remove beaver
   hosts: all
   tags:
     - rpc-pre-upgrades
+    - remove-beaver
   tasks:
     - name: Check if beaver is installed
       stat:
         path: /etc/init.d/beaver
       register: beaver_init_script
 
-    - name: Restart beaver if installed
+    - name: Make sure beaver is stopped
       service:
         name: beaver
-        state: restarted
+        enabled: no
+        state: stopped
       when: beaver_init_script.stat.exists
+
+    - name: Remove beaver etc files
+      file:
+        path: /etc/beaver
+        state: absent
+
+    - name: Remove beaver init files
+      file:
+        path: /etc/init.d/beaver
+        state: absent
+
+    - name: Remove beaver logrotate files
+      file:
+        path: /etc/logrotate.d/beaver
+        state: absent
+
+    - name: Remove beaver log files
+      file:
+        path: /var/log/beaver
+        state: absent
+
+    - name: Check if beaver is installed
+      shell: pip show -q Beaver
+      register: is_beaver_installed
+
+    - name: Uninstall beaver pip package
+      pip:
+        name: beaver
+        state: absent
+      when: is_beaver_installed.rc == 0
+
```

Comment:
OK I just tested this properly and well, it seems good old pip in my mac (crappy old 1.5.6) behaves differently to the version we're using (8.1.2), so I'm wrong about the return code always being 0.

And looking again to see if just letting the pip module do the right thing works, I can see the errors you were running in to. It seems the pip module is somewhat busted in ansible 1.9.4 (we checked, and it's fixed in 1.9.5 but breaks some lxc container stuff), in that it will error with an rc of 1 if the package is not installed.

Chatting with @evrardjp and @odyssey4me the best way we can handle this cleanly is to do something like (pseudo code):

failed_when stderr does not contain the string 'not installed' and stdout does not contain the string 'Successfully uninstalled'